core/src/main/scala/com/mesosphere/usi/models/SpecEvent.scala

```diff
@@ -0,0 +1,21 @@
+package com.mesosphere.usi.models
+
+import com.mesosphere.usi.core.models.{PodId, PodSpec, ReservationId}
+
+sealed trait SpecEvent
+
+/**
+  * Used to describe the initial state on which all SpecUpdated events will apply
+  * @param podSpecs
+  * @param reservationSpecs
+  */
+case class SpecsSnapshot(podSpecs: Seq[PodSpec], reservationSpecs: Seq[ReservationSpec]) extends SpecEvent
+
+/**
+  * Used to communicate that a specification was updated (added, changed, or deleted)
+  */
+sealed trait SpecUpdated extends SpecEvent
+
+case class PodSpecUpdated(id: PodId, newState: Option[PodSpec]) extends SpecUpdated
```

Comment:
So if the `newState == None` we infer that the podSpec was deleted? Wdyt about making it a `PodSpecDeleted(id: PodId)`?

Reply:
I prefer the Option

Reply:
But, but... `PodSpecDeleted` is so much clearer and easier to read.

Reply:
also, think about logging. grepping for all deleted podSpecs is easier with `PodSpecDeleted(...)` then it is with `PodSpecUpdated(???, None)`